libnavigation-base/src/main/java/com/mapbox/navigation/base/route/DirectionsSession.kt

```diff
@@ -16,5 +20,7 @@ interface DirectionsSession {
 
     interface RouteObserver {
         fun onRouteChanged(route: Route?)
```

Comment:
Look comment above re: `DirectionsResponse` / list of `DirectionsRoute`. Do we want to limit this API to only one `Route`?

Reply:
old api provide only one way(best one). We could provide all

Reply:
discussed, done

Reply:
`DirectionSession` is wrapper for `Route`, so if you'd like to receive all routes you should use `Route`'s implementation. cc @LukasPaczos

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/2240#discussion_r344964741

Reply:
This is what I was referring to (in action 😅) https://github.com/mapbox/mapbox-navigation-android/pull/2274#discussion_r349169264